### ISSUE TITLE
Invalid property specified for object of type plotly.graph_objs.layout.Legend: 'itemwidth'

### ISSUE THREAD

**Issue by bg-bi** [plotly/plotly.py#3178]
Despite being documented [here](https://plotly.com/python/reference/layout/#layout-legend-itemwidth), the `Figure.legend.itemwidth` property cannot be set in Python. I am using the latest version of Plotly.py at time of writing, v4.14.3. This issue is also discussed on the plotly community forum [here](https://community.plotly.com/t/issue-with-layout-legend-parameter-itemwidth/51363).

Here's a minimal example to reproduce the error:

```python
import plotly.graph_objects as go
fig = go.Figure()
fig.update_layout(
    legend=dict(itemwidth=20) # ValueError raised
)
```

**Comment by empet** [comment]
The minimum and the default value for `itemwidth` is 30,[https://github.com/plotly/plotly.js/blob/master/src/components/legend/attributes.js#L80](https://github.com/plotly/plotly.js/blob/master/src/components/legend/attributes.js#L80), but in your code it is set, 20.

**Comment by aykutaaykut** [comment]
I also have the same issue with itemwidth property even though I set it greater than 30 such as 50. I dont think the problem is with the value set, itemwidth is not recognized as a valid property.
`import plotly.graph_objects as go
fig = go.Figure()
fig.update_layout(
    legend=dict(itemwidth=50)
)

ValueError: Invalid property specified for object of type plotly.graph_objs.layout.Legend: 'itemwidth'
Did you mean "borderwidth"?`

**Comment by empet** [comment]
@aykutaaykut 
I referred to `itemwidth`  and it works, as you can see with this example:
```
import plotly.graph_objects as go
import numpy as np
fig = go.Figure(go.Scatter(x=np.arange(12), y= 2-3*np.random.rand(12)))
fig.update_layout(width=600, height=400,
legend_itemwidth=70, showlegend=True)
```
![itemwidth1](https://user-images.githubusercontent.com/3627253/116997390-66f8c600-ace5-11eb-99fc-1be965366927.png)

**Comment by aykutaaykut** [comment]
I figured out now, the problem was the plotly version. It was not working in v4.13 and it worked after I upgraded to v4.14.3.

**Comment by bg-bi** [comment]
My mistake, I did upgrade to v4.14.3, but it looks like I my environment didn't have plotly v4.14.3 at the time I tested this. I did however have the same issue as @aykutaaykut, the error showed as `Invalid property specified for object of type plotly.graph_objs.layout.Legend: 'itemwidth'` despite having a valid value set (like 50) so the error was confusing. The error message didn't show when my disappeared entirely after restarting my environment with the new version installed and activated.

Closing the issue.
